src/execution/operator/scan/physical_table_scan.cpp

```diff
@@ -108,7 +108,17 @@ SourceResultType PhysicalTableScan::GetData(ExecutionContext &context, DataChunk
 	if (g_state.in_out_final) {
 		function.in_out_function_final(context, data, chunk);
 	}
-	function.in_out_function(context, data, g_state.input_chunk, chunk);
+	switch (function.in_out_function(context, data, g_state.input_chunk, chunk)) {
+	case OperatorResultType::BLOCKED: {
+		auto guard = g_state.Lock();
+		return g_state.BlockSource(guard, input.interrupt_state);
```

Comment:
We discussed this, but I think our conclusion was wrong, this is fine to stay as is, you don't want to run `in_out_function_final` here

Reply:
Thanks for having a proper look!